sql/src/main/java/io/crate/planner/SubqueryPlanner.java

```diff
@@ -62,6 +62,18 @@ public Void visitSelectSymbol(SelectSymbol selectSymbol, Symbol parent) {
             return null;
         }
 
+        @Override
+        public Void visitFunction(io.crate.expression.symbol.Function function, Symbol context) {
+            for (Symbol argument : function.arguments()) {
+                process(argument, context);
+            }
+            var filter = (io.crate.expression.symbol.Function) function.filter();
```

Comment:
Why is the cast here necessary?

Would it make sense to move this into the `DefaultTraversalSymbolVisitor`, or does this break other visitors that depend on only args being processed?

Reply:
it is not necessary, dropping it and checking whether it is possible to move it to `DefaultTraversalSymbolVisitor`